management/server/http/policies_handler.go

```diff
@@ -170,23 +199,45 @@ func (h *Policies) CreatePolicy(w http.ResponseWriter, r *http.Request) {
 		for _, r := range req.Rules {
 			pr := server.PolicyRule{
 				ID:           xid.New().String(),
+				Name:         r.Name,
 				Destinations: groupMinimumsToStrings(account, r.Destinations),
 				Sources:      groupMinimumsToStrings(account, r.Sources),
-				Name:         r.Name,
+				Bidirect:     r.Bidirect,
 			}
+
 			pr.Enabled = r.Enabled
 			if r.Description != nil {
 				pr.Description = *r.Description
 			}
+
 			switch r.Action {
-			case api.PolicyRuleActionAccept:
+			case api.PolicyRuleUpdateActionAccept:
 				pr.Action = server.PolicyTrafficActionAccept
-			case api.PolicyRuleActionDrop:
+			case api.PolicyRuleUpdateActionDrop:
 				pr.Action = server.PolicyTrafficActionDrop
 			default:
 				util.WriteError(status.Errorf(status.InvalidArgument, "unknown action type"), w)
 				return
 			}
+
+			switch r.Protocol {
```

Comment:
these 2 switch statements could be in functions as the checks are the same as in update policy

Reply:
I did "extract" refactoring, so I moved the common code to `savePolicy` helper method, so there is no more duplicate code.